### ISSUE TITLE
Rework training flags/configuration so that it can capture a full training run including fine tuning passes

### ISSUE THREAD

**Issue by reuben** [mozilla/DeepSpeech#3476]
Currently the training code is configured via abseil-py command line flags, which can be put into a file and referenced with `--flagfile flags.txt`. Unfortunately CLI flags can get hard to read and manage as the number of specified options grows. We get many reports of errors due to incorrectly specified flags or shell syntax errors.

Another problem is that when specifying flags there is no scope. All flags exist on a global level and other than the brief documentation of each flag or the loose grouping of flags in `flags.py` the only way to actually know where the influence of a flag starts and ends is by reading and understanding the entire training code.

For complex training runs such as the ones used to create the English release models the flags can't capture the entire process, as it requires calling the training code multiple times with different flag values. The augmentation flags support a scheduling syntax so values can change in a controlled manner along the training run, but other flags such as learning rate and dropout rate don't.

The goal of this issue is to figure out a solution that allows training configuration to capture the full details of any training run. A related question is whether it is better to move from CLI flags to config files with a more readable/flexible syntax such as YAML.
